src/gui/guiFormSpecMenu.cpp

```diff
@@ -4184,8 +4184,10 @@ bool GUIFormSpecMenu::OnEvent(const SEvent& event)
 			m_client->makeScreenshot();
 		}
 
-		if (event.KeyInput.PressedDown && kp == getKeySetting("keymap_toggle_debug"))
+		if (event.KeyInput.PressedDown && kp == getKeySetting("keymap_toggle_debug") &&
+				(m_client != NULL && m_client->checkPrivilege("debug"))) {
 			m_show_debug = !m_show_debug;
```

Comment:
there's an edge case here: if the player had the "debug" privilege but it was revoked from him, debug mode will stay enabled

Reply:
@l-koehler it's not just about allowing them to disable it: If the priv is revoked, `m_show_debug` should always be set to `false`.

So the logic should go something like this: `if (m_client != NULL && !m_client->checkPrivilege("debug"))) m_show_debug = false; else if (... toggle? ...)  m_show_debug = !m_show_debug;`